packages/discord-bot/src/utils/praisePermissions.ts

```diff
@@ -0,0 +1,23 @@
+import { CommandInteraction } from 'discord.js';
+import { getSetting } from './settings';
+
+export const praiseAllowedInChannel = async (
+  interaction: CommandInteraction,
+  channelId: string
+): Promise<boolean> => {
+  const allowedInAllChannels = await getSetting(
+    'PRAISE_ALLOWED_IN_ALL_CHANNELS'
+  );
+  const allowedChannelsList = (await getSetting(
+    'PRAISE_ALLOWED_CHANNEL_IDS'
+  )) as String;
+  if (
+    allowedInAllChannels === 'false' &&
+    allowedChannelsList.indexOf(channelId) === -1
+  ) {
+    await interaction.editReply('Praise not allowed in this channel.');
```

Comment:
Format like other error messages: `**❌ ...**`. Consider refactoring and make a general `errorReply` function to be used by all commands.

Reply:
Q) Since other errors are based on format entered by user, should we add this text as a setting?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question&nbsp; | &nbsp;**Tags:** Brittle